File: works/tmp/1783.py
N, M = map(int, input().split())

if N <= 1:
    print(1)
elif N <= 2:
    if M >= 9: print(4)
    else: print((M-1) // 2 + 1)
elif M < 5:
    print(M)
elif M <= 5:
    print(M-1)
elif M > 5:
    print(M-2)

File: works/tmp/1783.cs
using static System.Console;
using System;
using System.Collections;
using System.Text;
using System.IO;
using System.Collections.Generic;

namespace Project1
{
    public class Class1
    {
       
        public static void Main(string[] args)
        {
            string[] s = ReadLine().Split(' ');
            int N = int.Parse(s[0]);
            int M = int.Parse(s[1]);

            if(N <= 1)
            {
                Write(1);
                return;
            }
            if(N <= 2)
            {
                if (M >= 9) Write(4);
                else Write((M - 1) / 2 + 1);
                return;
            }

            if(M < 5)
            {
                Write(M);
                return;
            }
            if(M <= 5)
            {
                Write(M - 1);
                return;
            }

            if(M > 5)
            {
                Write(M - 2);
            }
        }
    }
}
